client/src/controllers/UpgradeController.ts

```diff
@@ -0,0 +1,84 @@
+import { Controller } from '@hotwired/stimulus';
+import { VersionNumber, VersionDeltaType } from '../utils/version';
+
+export class UpgradeController extends Controller<HTMLElement> {
+  static classes = ['hidden'];
+  static targets = ['latestVersion', 'link'];
+  static values = {
+    currentVersion: String,
+    ltsOnly: { default: false, type: Boolean },
+    url: { default: 'https://releases.wagtail.org/latest.txt', type: String },
+  };
+
+  currentVersionValue: string;
+  hiddenClass: string;
+  ltsOnlyValue: any;
+  latestVersionTarget: HTMLElement;
+  linkTarget: HTMLElement;
+  urlValue: string;
+
+  connect() {
+    this.checkVersion();
+  }
+
+  checkVersion() {
+    const releasesUrl = this.urlValue;
+    const currentVersion = new VersionNumber(this.currentVersionValue);
+    const showLTSOnly = this.ltsOnlyValue;
+    fetch(releasesUrl, {
+      referrerPolicy: 'strict-origin-when-cross-origin',
+    })
+      .then((response) => {
+        if (response.status !== 200) {
+          // eslint-disable-next-line no-console
```

Comment:
We could probably use Stimulus' error callbacks here - avoids us calling console.error directly 

Double check but I'm pretty sure you can just throw an error and Stimulus will just handle this Async and push it to logging if needed. It won't break the whole JS UI - just stop the specific controller method from being run. 

```
throw Error('Unexpexted...');
```

See
- https://github.com/hotwired/stimulus/blob/main/src/core/application.ts#L87
- https://stimulus.hotwired.dev/handbook/installing#error-handling

Reply:
Hi @lb- 
I would need to write a new error test for this right?
The handleError function only comes to play when we are displaying the error in the DOM?